src/PluginsSystem/Tools/Microsoft.Performance.Toolkit.Plugins.Cli/Commands/Common/PackGenCommonCommand.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Performance.Toolkit.Plugins.Cli.Manifest;
+using Microsoft.Performance.Toolkit.Plugins.Cli.Processing;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Cli.Commands.Common
+{
+    internal abstract class PackGenCommonCommand<TArgs>
+        : ICommand<TArgs>
+        where TArgs : PackGenCommonArgs
+    {
+        protected readonly IManifestLocatorFactory manifestLocatorFactory;
+        protected readonly IPluginArtifactsProcessor artifactsProcessor;
+        protected readonly ILogger<PackGenCommonCommand<TArgs>> logger;
+
+        protected PackGenCommonCommand(
+            IManifestLocatorFactory manifestLocatorFactory,
+            IPluginArtifactsProcessor artifactsProcessor,
+            ILogger<PackGenCommonCommand<TArgs>> logger)
+        {
+            this.manifestLocatorFactory = manifestLocatorFactory;
+            this.artifactsProcessor = artifactsProcessor;
+            this.logger = logger;
+        }
+
+        public int Run(TArgs args)
+        {
+            if (!TryGetProcessedPluginResult(args, out ProcessedPluginResult? processedSource))
+            {
+                return 1;
+            }
+
+            return RunCore(args, processedSource!);
```

Comment:
another nit:

Instead of using `!` everywhere, we can decorate `out` parameters with https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.notnullwhenattribute?view=net-7.0